src/edu/washington/escience/myria/operator/network/distribute/BroadcastDistributeFunction.java

```diff
@@ -0,0 +1,36 @@
+package edu.washington.escience.myria.operator.network.distribute;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+
+import edu.washington.escience.myria.util.MyriaArrayUtils;
+
+/**
+ * Broadcast distribute function.
+ */
+public final class BroadcastDistributeFunction extends DistributeFunction {
+
+  /** Required for Java serialization. */
+  private static final long serialVersionUID = 1L;
+
+  /**
+   */
+  @JsonCreator
+  public BroadcastDistributeFunction() {
+    super(new SinglePartitionFunction(), null);
+  }
+
+  /**
+   * @param numDestinations number of destination
+   */
+  public BroadcastDistributeFunction(final int numDestinations) {
```

Comment:
My understanding of Broadcast is that it duplicates a tuple to all the workers, so we don't need a list of them, while the mappings from partitions to destinations of a multiway join is manually specified in its `cellPartitition` so that Broadcast is not the correct one to use. However, if we want to extend `Broadcast` to broadcast to a subset of workers then yes we may need that, and I can envision how `HyperCubeShuffle` uses it.